spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1199,3 +1199,137 @@ The following metrics are produced for each channel declared by the application
 // ==== Custom message types
 
 // Implementations may provide custom subtypes of `Message` as an end user API, for the purpose of providing additional metadata associated with the messaging provider. If an application declares that it consumes or produces a subtype of the `Message` wrapper, the implementation should validate that the messaging provider supports that message wrapper type, and raise a deployment exception if it doesn't.
+
+== Publishing messages to a channel from imperative code
+
+Traditionally, the reactive world and imperative world are separated and operate in parallel. Reactive Messaging deals with streams of data in the reactive world, while the imperative world is pretty much point to point and synchronous communication. However, imperative programme sometimes needs to connect to reactive streams so that responses can be emitted to a destination service. 
+Bridging the two worlds is very valuable thing to do, so that one microservice can use technologies from both environments. For an instance, a JAX-RS resource might want to publish messages to a Reactive Messaging channel. This section is about enabling imperative code to publish messages to a Reactive Messaging channel, so that it can be consumed by a consumer.
+
+This is achieved by using `@Channel`.
+
+=== Sending messages to a channel
+
+In order to publish messages on a _channel_, we need to connect a channel first. `@Channel("channel_name")` is used for this purpose.
+`@Channel` is a qualifier to inject a _channel_ to a CDI bean or any class that supports CDI injection. 
```

Comment:
"Connect a channel" doesn't read right. You could connect to a channel, or connect something else to a channel.

Also, a "channel" isn't something you inject. You either inject a `Publisher`, `PublisherBuilder` or `Emitter`.

So maybe remove these two lines and incorporate this information into the sections below:

 You can either inject a `Publisher` or `PublisherBuilder` to receive messages from a specified channel, as shown below. You must use the `@Channel` qualifier to specify the name of the channel.

You can inject an `Emitter` and use it to send either payloads (`X`) or messages (`Message<X>`) to a channel as demonstrated below. When injecting the `Emitter` you must use the `@Channel` qualifier to specify the name of the channel.

Reply:
Thanks @Azquelt ! reworded with your help.